test/onnx/onnx_test.cpp

```diff
@@ -2099,6 +2099,24 @@ TEST_CASE(gathernd_test)
     EXPECT(p == prog);
 }
 
+TEST_CASE(gathernd_dyn_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    auto l0  = mm->add_parameter("data",
+                                migraphx::shape{migraphx::shape::float_type, {{2, 2, 2}, {2, 2}}});
+    auto l1  = mm->add_parameter("indices",
+                                migraphx::shape{migraphx::shape::int64_type, {{2, 2}, {2, 2}}});
+    auto r   = mm->add_instruction(migraphx::make_op("gathernd"), l0, l1);
+    mm->add_return({r});
+
+    migraphx::onnx_options options;
+    options.map_dyn_input_dims["data"]    = {{2, 2, 2}, {2, 2}};
+    options.map_dyn_input_dims["indices"] = {{2, 2}, {2, 2}};
```

Comment:
This test is not dynamic, it has all fixed `dynamic_dimensions`. I think it works because `map_dyn_input_dims` overrides to a `dynamic_dimension` even if it's fixed.

Reply:
Isn't the default dimension typing a functionality that also needs to be tested?

As for fixed vs. non-fixed dimensions, they all use the dynamic_dimension type.  This test does force the code to get the mins, the maxes, and the opts right.  Is there something else about non-fixed dimensions that needs to be checked at this point?  If so, maybe we should clarify what we want to test for.

Also, should we include multiple tests, including fixed and non-fixed dynamic_dimensions?  Does that add any value, or is it overkill?

Reply:
I don't understand the first question.

The tests for the output_shape should be in op_shape_test.cpp. If we only test when the dynamic_dimension is fixed we won't know if it works when it's not fixed.

Testing all fixed dynamic_dimensions doens't make much sense because it's equivalent to a static shape at that point.

Reply:
Changed several dimensions in several tests to be non-fixed.  

We agreed that adding the functionality  of checking (and testing) for invalid dynamic dimension values may be a good idea but would lie outside the scope of this operator and its tests.